src/components/ReportActionItem/MoneyRequestView.tsx

```diff
@@ -170,6 +171,19 @@ function MoneyRequestView({
 
     let amountDescription = `${translate('iou.amount')}`;
 
+    const hasRoute = TransactionUtils.hasRoute(transaction);
+    const distance = transaction?.routes?.route0?.distance ?? 0;
```

Comment:
The distance for an existing transaction should come from `transaction.comment.customUnit.quantity`

Reply:
@neil-marcellini why? transaction.comment.customUnit.quantity is empty right after the transaction is made (and 0.00 is displayed for the distance) and we would need to change the logic for `getDistanceForDisplay` which currently uses `getRoundedDistanceInUnits` which accepts `distanceInMeters`. `getRoundedDistanceInUnits` was also previously used this way for merchant.

Reply:
The distance is also stored on the transaction merchant of course, but the place I mentioned feels like a more direct way to access it.

Reply:
Oh, I see what you're saying now. `transaction.comment.customUnit.quantity` is empty right after the request is created and before the response comes back from the backend. Maybe the solution is to save it into the optimistic data?

Reply:
ok, but still we would need to change the current logic for getRoundedDistanceInUnits which accepts distanceInMeters - quantity is saved in the unit it was created? So we would need to detect whether it's mile or km

Reply:
> quantity is saved in the unit it was created?

Yes. Yeah I suppose some changes will be needed. I'm noticing that we would need to get the unit from the personal policy custom unit, but it doesn't seem to be present. I could change the backend to send that, or we could extract the miles text from the merchant directly.

I think getting it from the merchant might be the most easy thing to do right now. What do you think?

Reply:
why can't we get it from the route? in the confirmation, we do it this way:

`const distance = lodashGet(transaction, 'routes.route0.distance', 0);`

That's why I followed the same pattern here. I didn't have to change any of the previously added utils and this value was accessible right after the request

Reply:
I think the difference is that the confirmation page is only visible while creating a new request. If you get the distance from the route, how would it work in this case?

1. Create a p2p distance request
2. Sign out
3. Sign back in
4. View the distance on the MoneyRequestView page

At step 4 the route won't exist. You're welcome to do whatever is easiest as long as it follows our existing patterns/philosophies, and works in all cases.

Reply:
ok, so I'm now extracting the distance from the merchant. Adding it to optimistic data would require a lot of digging